pkg/action/executor/move_util.go

```diff
@@ -200,74 +203,71 @@ func movePod(clusterScraper *cluster.ClusterScraper, pod *api.Pod, nodeName, par
-			defer func() {
-				glog.V(3).Infof("Unpausing pods controller: %s for pod: %s", gPKind, podQualifiedName)
-				// We conservatively try additional 3 times in case of any failure as we don't
-				// want the parent to be left in a bad state.
-				err := commonutil.RetryDuring(DefaultExecutionRetry, DefaultRetryShortTimeout,
-					DefaultRetrySleepInterval, func() error {
-						return ResourceRollout(gPClient, grandParent, unpause)
-					})
-				if err != nil {
-					glog.Errorf("Move pod warning: %v", err)
-				}
-			}()
-
-			// step B2:
-			glog.V(3).Infof("Pausing pods controller: %s for pod: %s", gPKind, podQualifiedName)
-			err := ResourceRollout(gPClient, grandParent, pause)
-			if err != nil {
-				return nil, err
-			}
-		}
-
-		podList, err = parentsPods(parent, podClient)
-		if err != nil {
-			return nil, err
-		}
-
-		valid := true
-		invalid := false
-		// step B7: (via defer)
+	if grandParent != nil {
+		pause := true
+		unpause := false
+		// step 8: (via defer)
 		defer func() {
-			glog.V(3).Infof("Updating scheduler of %s for pod %s back to default.", parentKind, podQualifiedName)
+			glog.V(3).Infof("Unpausing pods controller: %s for pod: %s", gPKind, podQualifiedName)
 			// We conservatively try additional 3 times in case of any failure as we don't
 			// want the parent to be left in a bad state.
 			err := commonutil.RetryDuring(DefaultExecutionRetry, DefaultRetryShortTimeout,
 				DefaultRetrySleepInterval, func() error {
-					return ChangeScheduler(pClient, parent, valid)
+					return ResourceRollout(gPClient, grandParent, unpause)
 				})
 			if err != nil {
 				glog.Errorf("Move pod warning: %v", err)
 			}
-
 		}()
-		// step B3:
-		glog.V(3).Infof("Updating scheduler of %s for pod %s to unknown (turbo-scheduler).", parentKind, pod.Name)
-		err = ChangeScheduler(pClient, parent, invalid)
+
+		// step 2:
+		glog.V(3).Infof("Pausing pods controller: %s for pod: %s", gPKind, podQualifiedName)
+		err := ResourceRollout(gPClient, grandParent, pause)
 		if err != nil {
 			return nil, err
 		}
+	}
 
-		// step A3/B4:
+	podList, err = parentsPods(parent, podClient)
+	if err != nil {
+		return nil, err
+	}
+
+	valid := true
+	invalid := false
+	// step 7: (via defer)
+	defer func() {
+		glog.V(3).Infof("Updating scheduler of %s for pod %s back to default.", parentKind, podQualifiedName)
+		// We conservatively try additional 3 times in case of any failure as we don't
+		// want the parent to be left in a bad state.
+		err := commonutil.RetryDuring(DefaultExecutionRetry, DefaultRetryShortTimeout,
+			DefaultRetrySleepInterval, func() error {
+				return ChangeScheduler(pClient, parent, valid)
+			})
+		if err != nil {
+			glog.Errorf("Move pod warning: %v", err)
+		}
+
+	}()
+	// step 3:
+	glog.V(3).Infof("Updating scheduler of %s for pod %s to unknown (turbo-scheduler).", parentKind, pod.Name)
+	err = ChangeScheduler(pClient, parent, invalid)
+	if err != nil {
+		return nil, err
+	}
+
+	if podUsingVolume {
```

Comment:
Not necessary to be in this review, but instead of simply check if podUsingVolume, can we check if the access mode is readWriteOnce? Meaning if it's actually ReadWriteMany, shall we also start and delete? Can we test and see if it's working?

Reply:
That makes sense @enlinxu, and I think it can be done too. `multi-attach` should work if the mode is `readWriteMany` and underlying volume supports the same. I will create a story to have this implemented.

Reply:
@enlinxu  https://vmturbo.atlassian.net/browse/OM-84927